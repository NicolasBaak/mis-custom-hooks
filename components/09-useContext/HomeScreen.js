import React, {useContext} from 'react';
import { UserContext } from './UserContex';
export const HomeScreen = () => {

  const {user, setUser} = useContext(UserContext);
  console.log(user);

  const handleClick = ()=>{
    setUser({});
  }
  return (
  <>
    <h1>HomeScreen</h1>
    <hr/>
    <pre className=''>
      {JSON.stringify(user, null, 3)}
    </pre>

    <button
      className='btn btn-primary'
      onClick={handleClick}
    >
      LogOut
    </button>
  </>
  );
};
