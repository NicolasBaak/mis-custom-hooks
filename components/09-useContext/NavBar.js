import React from 'react';
import { NavLink, Link } from 'react-router-dom';

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                
                <Link to="/" className="navbar-brand">useContext</Link>
               
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav">
                        <li className="nav-item">
                            <NavLink to="/" activeclassname="active" className="nav-link" aria-current="page" >Home</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink to="/about" activeclassname="active" className="nav-link" >About</NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink to="/loginscreen" activeclassname="active" className="nav-link" >Login</NavLink>
                        </li>
                    </ul>
                </div>
            </div>
        </nav>
  );
};
