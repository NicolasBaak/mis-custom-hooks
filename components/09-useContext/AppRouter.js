import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from "react-router-dom";

import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen'
import { AboutScreen } from './AboutScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
  return (
      <Router>
          <div >
              <NavBar />

              <div className='container'>
       
               {/**<Routes></Routes> Nos ayuda a mostrar rutas */}
               <Routes>
                   <Route path="/" element={<HomeScreen/>}/>
                   <Route path="/about" element={<AboutScreen/>}/>
                   <Route path="/loginscreen" element={<LoginScreen/>}/>
                   
                   {/* Si no se encuentra la ruta va al homeScreen, tambien podria definirse un 404page */}
                   <Route path='*' element={<HomeScreen />} />
               </Routes>
            
               </div>
            </div>
      </Router>
  );
};
