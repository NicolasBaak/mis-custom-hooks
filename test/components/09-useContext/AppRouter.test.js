import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContex';

describe('Pruebas en <AppRouter />', () => { 

    const user = {
        id: 1,
        name: 'Nicolas'
    }
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('Debe mostrarse correctamente', () => {  
        expect(wrapper).toMatchSnapshot();
    })
 })