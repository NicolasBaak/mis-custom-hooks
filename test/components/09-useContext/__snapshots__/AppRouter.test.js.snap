// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              activeclassname="active"
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              activeclassname="active"
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              activeclassname="active"
              className="nav-link"
              href="/loginscreen"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <hr />,
      <pre
        className=""
      >
        {
   "id": 1,
   "name": "Nicolas"
}
      </pre>,
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        LogOut
      </button>,
    ]
  </div>
</div>
`;
